### ISSUE TITLE
Split-off the internal server routes into a new "internal-server" component

### ISSUE THREAD

**Issue by fniessink** [ICTU/quality-time#2961]
Blocked by #2960.

Database initialization remains in the server component.

The notifier and collector use the internal server exclusively. We accept that this causes a small bit of duplication between the server and internal server.

The internal server also uses bottle (for the time being; replacing bottle with FastAPI is out of scope of this issue).

Don't forget to update the developer documentation (including the dot graph).

This issue results in a MAJOR version upgrade: operators need to update their docker composition.

Mikado steps (merge to master after every step):
 
- [x] 1. Move logos from server/src/external/routes to server/src/external/data_model
- [x] 2. Move server/src/external/data_model to shared_python/src/shared/data_model
- [x] 3. Rename server component to external_server (including relevant filenames)
- [x] 4. Remove the external package in the external_server and move the contents up.
- [x] 5. Add /api-internal/v3/report endpoint to server/src/internal/routes
- [x] 6. Add /api-internal/v3/datamodel endpoint to server/src/internal/routes
- [x] 7. Move the internal initialization that is still in the server/src/external/initialization package to a new server/src/internal/initialization package (consider moving the modules in the initialization package one by one). 
- [x] 8. In components/collector/tests/data_model_fixture.py and components/notifier/tests/data_model.py simply import the data_model from shared_python and remove the PATH manipulation
- [x] 9. In the docs (https://github.com/ICTU/quality-time/blob/f9df7c7f184d9bc1cd2c1403a7af85c1da39a2d3/docs/src/create_metrics_and_sources_md.py#L20), simply import the data_model from shared_python and remove the PATH manipulation
- [x] 10. Update feature tests to use the internal server
- [x] 11. Remove internal-server code from external_server/src/internal
- [x] 12. Update development documentation
- [x] 13. Update software documentation
- [x] 14. Update deployment documentation
- [x] 15. Add health endpoint to internal server
- [x] 16. Update release notes
- [x] 17. Release a 4.0.0 release candidate and run this somewhere for an extended period
- [x] 18. Test a rollback to version 3.x
